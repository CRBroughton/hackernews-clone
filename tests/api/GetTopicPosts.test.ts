// tests/Post.test.ts
import { createTestContext } from '../__helpers'
const ctx = createTestContext()

it('Gets posts from the a specific user', async() => {
  const feed = await ctx.client.request(`
      query {
        getTopicPosts(topic: "General") {
          id
          description
          url
          topic
          postedBy {
            id
            name
            email
          }
          voters {
            id
            name
            email
          }
        }
      }
    `)
  expect(feed).toMatchSnapshot()
})
