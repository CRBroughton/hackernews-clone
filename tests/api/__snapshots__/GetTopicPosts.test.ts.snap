// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets posts from the a specific user 1`] = `
Object {
  "getTopicPosts": Array [
    Object {
      "description": "Vite.js",
      "id": "040688e1-c08f-4752-8d5f-888ffb36ef02",
      "postedBy": Object {
        "email": "alice@prisma.io",
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "topic": "General",
      "url": "https://vitejs.dev",
      "voters": Array [],
    },
    Object {
      "description": "My Portfolio",
      "id": "2df7837b-788d-4410-aa65-919da54f7175",
      "postedBy": Object {
        "email": "alice@prisma.io",
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "topic": "General",
      "url": "https://crbroughton.me",
      "voters": Array [],
    },
    Object {
      "description": "Code-First GraphQL schemas for JavaScript/TypeScript",
      "id": "16e11628-462a-4235-afb2-21c0db7425ba",
      "postedBy": Object {
        "email": "alice@prisma.io",
        "id": "97527cbe-2de0-40cd-a953-8caae780e66e",
        "name": "Alice",
      },
      "topic": "General",
      "url": "https://nexusjs.org",
      "voters": Array [],
    },
  ],
}
`;
